src/socket/address.cr

```diff
@@ -728,6 +774,24 @@ class Socket
       sockaddr.as(LibC::Sockaddr*)
     end
 
+    # Returns the interface name for a scoped/zoned link-local IPv6 address.
+    # This only works on properly initialized link-local IPv6 address objects.
+    # In any other case this will return nil.
+    #
+    # The LibC structs track the zone via a numerical interface index as
+    # enumerated by the kernel. To keep our abstraction class in line, we
+    # also only keep the interface index around.
+    #
+    # This helper method exists to look up the interface name based on the
+    # associated zone_id property.
+    def link_local_interface : String | Nil
+      return nil if @zone_id.zero?
+      return nil unless (@family == Socket::Family::INET6 && link_local?)
+      buf = uninitialized StaticArray(UInt8, LibC::IF_NAMESIZE)
+      LibC.if_indextoname(@zone_id, buf)
+      String.new(buf.to_unsafe)
```

Comment:
This might need to be extracted as a `Crystal::System::Socket.link_local_interface(zone_id)` method. Win32 happens to use the same API as POSIX, but WASI doesn't implement this for example, and should raise a `NotImplementedError` exception.

Reply:
You're suggesting to move that helper method out of `Socket::Address` into the `Crystal::System::Socket` module?
We could also add a macro to the beginning of the method, e.g.:
```crystal
    def link_local_interface : String | Nil
      {% if flag?(:wasm32) %}
        raise NotImplementedError.new "Socket::Address.link_local_interface"
      {% end %}
```

Reply:
I have added the macro now, please let me know if you want this implemented differently.

Reply:
adjusted to
```crystal
{% unless LibC.has_method?(:if_nametoindex) %}
```